File: 2903-insert-greatest-common-divisors-in-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode insertGreatestCommonDivisors(ListNode head) {
        ListNode temp = head;
        while(temp.next!=null){
            ListNode ne = new ListNode(gcd(temp.val,temp.next.val));
            ne.next = temp.next;
            temp.next = ne;
            temp = ne.next;
        }
        return head;
    }
    int gcd(int a,int b){
        for(int i=Math.min(a,b);i>0;i--){
            if(a%i==0 && b%i==0)
                return i;
        }
        return 1;
    }
}

File: 2903-insert-greatest-common-divisors-in-linked-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp=head
        while temp and temp.next:
            x,y = temp.val,temp.next.val
            new = ListNode(math.gcd(x,y))
            new.next=temp.next
            temp.next=new
            temp=new.next
        return head

        
